docs/design-documents/hal/0002-pinmap-extension.md

```diff
@@ -0,0 +1,83 @@
+# HAL PinMap design document
+
+# Table of contents
+
+- [Description](#Description)
+- [Motivation](#Motivation)
+- [Requirements](#Requirements)
+- [Design choices](#Design-choices)
+- [Questions](#Questions)
+
+<!-- tocstop -->
+
+# Introduction
+
+Adds board introspection capabilities and tools.  
+These tools will help designing better tests for the HAL.
+
+### Overview and background
+
+At the time being, drivers are only tested on a single "default" peripheral. However, some target feature the same peripheral through different blocks' implementations for example the SPI may be provided on a single MCU by its USART, QSPI and SSP peripherals.
```

Comment:
```suggestion
At the time being, drivers are only tested on a single "default" peripheral. However, some targets feature the same peripheral through different blocks' implementations for example the SPI may be provided on a single MCU by its USART, QSPI and SSP peripherals.
```

though, the entire second sentence feels like it could use some punctuation.